cli/dcoscli/node/main.py

```diff
@@ -475,34 +476,61 @@ def _log(follow, lines, leader, slave):
     :type leader: bool
     :param slave: the slave ID to print
     :type slave: str | None
+    :param component: DC/OS component name
+    :type component: string
+    :param filters: a list of filters ["key:value", ...]
+    :type filters: list
     :returns: process return code
     :rtype: int
     """
 
     if not (leader or slave):
-        raise DCOSException('You must choose one of --leader or --mesos-id.')
+        raise DCOSException(
+            'You must choose one of --leader or --mesos-id.')
 
     if lines is None:
         lines = 10
 
     lines = util.parse_int(lines)
 
     try:
-        _dcos_log(follow, lines, leader, slave)
+        _dcos_log(follow, lines, leader, slave, component, filters)
         return 0
     except DCOSException as e:
         emitter.publish(DefaultError(e))
         emitter.publish(DefaultError('Falling back to files API...'))
     except Exception:
         raise
 
+    if component or filters:
+        raise DCOSException('--component or --filter is not '
+                            'supported by files API')
+
     # fail back to mesos files API.
     mesos_files = _mesos_files(leader, slave)
     log.log_files(mesos_files, follow, lines)
     return 0
 
 
-def _dcos_log(follow, lines, leader, slave):
+def _get_unit_type(unit_name):
+    """ Get the full unit name including the type postfix
+        or default to service
+    """
+    if not unit_name:
+        raise DCOSException('Empty systemd unit parameter')
+
+    # https://www.freedesktop.org/software/systemd/man/systemd.unit.html
+    unit_types = ['service', 'socket', 'device', 'mount', 'automount',
+                  'swap', 'target', 'path', 'timer', 'slice', 'scope']
+
+    for unit_type in unit_types:
+        if unit_name.endswith('.{}'.format(unit_type)):
+            return unit_name
```

Comment:
the idea of the function is to default to a `name.service`
for example user can provide:
`dcos node log --leader --component dcos-3dt`
and it should be equal to `dcos node log --leader --component dcos-3dt.service`

at the same time user can query
`dcos node log --leader --component dcos-signal.timer`